tests/integration/models/release.spec.js

```diff
@@ -286,6 +286,66 @@ describe('Release Model', function () {
 			});
 		});
 
+		parallel('balena.model.release.finalize()', function () {
+			const createDraftRelease = async function (commit) {
+				const userId = await balena.auth.getUserId();
+				return await balena.pine.post<BalenaSdk.Release>({
+					resource: 'release',
+					body: {
+						belongs_to__application: this.application.id,
+						is_created_by__user: userId,
+						commit: commit,
+						status: 'success',
+						source: 'cloud',
+						is_final: false,
+						composition: '{}',
+						start_timestamp: new Date().toISOString(),
+					},
+				});
+			};
+
+			it('should finalize a release by id', function () {
+				return createDraftRelease('abcdefg').then(function (draftRelease) {
+					return balena.models.release
+						.finalize(draftRelease.id)
+						.then(async function () {
+							const finalRelease = await balena.models.release.get(
+								draftRelease.id,
+							);
+							await balena.pine.delete({
+								resource: 'release',
+								id: draftRelease.id,
+							});
+							expect(finalRelease).to.deep.match({
+								commit: 'abcdefg',
+								id: draftRelease.id,
+								is_final: true,
+							});
+						});
+				});
+			});
+			it('should finalize a release by commit', function () {
+				return createDraftRelease('efghijklmnop').then(function (draftRelease) {
+					return balena.models.release
+						.finalize('efghi')
+						.then(async function () {
+							const finalRelease = await balena.models.release.get(
+								draftRelease.id,
+							);
+							await balena.pine.delete({
+								resource: 'release',
+								id: draftRelease.id,
+							});
+							expect(finalRelease).to.deep.match({
+								commit: 'efghijklmnop',
+								id: draftRelease.id,
+								is_final: true,
+							});
+						});
+				});
+			});
+		});
```

Comment:
How about the following ?
```suggestion
		describe('balena.model.release.finalize()', function () {
			const fields = ['id', 'commit'] as const;
			const testReleaseByField: Dictionary<BalenaSdk.Release> = {};
			const createDraftRelease = async function (commit) {
				const userId = await balena.auth.getUserId();
				return await balena.pine.post<BalenaSdk.Release>({
					resource: 'release',
					body: {
						belongs_to__application: this.application.id,
						is_created_by__user: userId,
						commit: commit,
						status: 'success',
						source: 'cloud',
						is_final: false,
						composition: '{}',
						start_timestamp: new Date().toISOString(),
					},
				});
			};

			before(async function () {
				let i = 0;
				await Promise.all(fields.map(async (field, i)) => {
					testReleaseByField[field] = await createDraftRelease(`abcdef${i}`);
				});
			});

			after(async function () {
				await balena.pine.delete({
					resource: 'release',
					options: {
						$filter: {
							id: { $in: Object.values(testReleaseIdByField) }
						}
					}
				});
			})

			parallel('', function () {
				fields.forEach((field) => {
					it(`should finalize a release by ${field}`, async function () {
						const draftRelease = testReleaseIdByField[field];
						await balena.models.release .finalize(draftRelease[field]);

						const finalRelease = await balena.models.release.get(
							draftRelease.id,
						);
						expect(finalRelease).to.deep.match({
							id: draftRelease.id,
							commit: draftRelease.commit,
							is_final: true,
						});
					});
				});
			});
		});
```
Afair we need the extra `decribe` b/c `parallel` is a little picky with `before`s & `after`s.

Reply:
I took a slightly different approach because my first way of doing things was breaking other tests.  Tell me what you think of it now.

Reply:
@toochevere in general we try to keep the number of app creation-deletions to a minimum, since that's a slow operation atm in the API.
The problem with deleting the release was that the release was still the target release of the application when trying to delete it, and that's why you needed to delete the whole app.

I moved this at the end of the above `describe()` section, just before the app gets deleted.
If you like it, then feel free to squash the commit with yours and we can get this merged.